packages/core/src/interpreter.ts

```diff
@@ -668,164 +539,22 @@ export class Interpreter<
       child.send(event);
     }
   }
-  private defer(sendAction: SendActionObject): void {
+  // TODO: make private (and figure out a way to do this within the machine)
+  public defer(sendAction: SendActionObject): void {
     this.delayedEventsMap[sendAction.params.id] = this.clock.setTimeout(() => {
       if (sendAction.params.to) {
-        this.sendTo(sendAction.params._event, sendAction.params.to);
+        sendAction.params.to.send(sendAction.params._event);
       } else {
         this.send(sendAction.params._event as SCXML.Event<TEvent>);
       }
     }, sendAction.params.delay as number);
   }
-  private cancel(sendId: string | number): void {
+
+  // TODO: make private (and figure out a way to do this within the machine)
```

Comment:
The timers could potentially be kept in the internal state. However, I don't have a clear picture of how that would work exactly.

Reply:
Actually, maybe this shouldn't be moved to the machine at all. If we remove this stuff from here then it won't make sense to have `clock` as one of the Interpreter's options.